---
layout: default
title: Publications
modified: 6/2/18
description: Papers, talks, and posters about GRNsight, a web application and service for visualizing models of small- to medium-scale gene regulatory networks.
---
            <p id="pageTitle">Publications</p>

            <div id="pageContent">
                <div class="panel-group">
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <p class="panel-title aboutHeaders">
                                <a href="#papers" data-toggle="collapse">
                                    Papers
                                </a>
                            </p>
                        </div>

                        <div id="papers" class="panel-collapse collapse in">
                            <div class="panel-body">
                                
                                <p class="publication">
                                    <b>September 12, 2016</b><br>
                                    Kam D. Dahlquist, John David N. Dionisio, Ben G. Fitzpatrick, Nicole A. Anguiano, Anindita Varshneya, Britain J. Southwick, Mihir Samdarshi. (2016) GRNsight: a web application and service for visualizing models of small- to medium-scale gene regulatory networks. <em>PeerJ Computer Science</em> 2:e85 <a href="https://doi.org/10.7717/peerj-cs.85" target="_blank">https://doi.org/10.7717/peerj-cs.85</a><br><br>

                                    <a href="https://doi.org/10.7717/peerj-cs.85" target="_blank">View HTML with embedded PDF</a><br>
                                    <a href="https://peerj.com/articles/cs-85.pdf" target="_blank">Download PDF (2.7 MB)</a><br>
                                    <a href="https://peerj.com/articles/cs-85/reviews/" target="_blank">See peer review history</a><br><br>
                                    
                                    <i>Note that preprints of this article were posted in PeerJ Preprints on <a href="https://peerj.com/preprints/2068v1/" target="_blank">May 23</a> and <a href="https://peerj.com/preprints/2068v2/" target="_blank">August 18</a>, 2016.</i><br><br>

                                    <b>Abstract</b><br>
                                    GRNsight is a web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. The original motivation came from our efforts to perform parameter estimation and forward simulation of the dynamics of a differential equations model of a small GRN with 21 nodes and 31 edges. We wanted a quick and easy way to visualize the weight parameters from the model which represent the direction and magnitude of the influence of a transcription factor on its target gene, so we created GRNsight. GRNsight automatically lays out either an unweighted or weighted network graph based on an Excel spreadsheet containing an adjacency matrix where regulators are named in the columns and target genes in the rows, a Simple Interaction Format (SIF) text file, or a GraphML XML file. When a user uploads an input file specifying an unweighted network, GRNsight automatically lays out the graph using black lines and pointed arrowheads.  For a weighted network, GRNsight uses pointed and blunt arrowheads, and colors the edges and adjusts their thicknesses based on the sign (positive for activation or negative for repression) and magnitude of the weight parameter. GRNsight is written in JavaScript, with diagrams facilitated by D3.js, a data visualization library. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized to support the specific needs of GRNs. Nodes are rectangular and support gene labels of up to 12 characters.  The edges are arcs, which become straight lines when the nodes are close together.  Self-regulatory edges are indicated by a loop.  When a user mouses over an edge, the numerical value of the weight parameter is displayed. Visualizations can be modified by sliders that adjust the force graph layout parameters and through manual node dragging. GRNsight is best-suited for visualizing networks of fewer than 35 nodes and 70 edges, although it accepts networks of up to 75 nodes or 150 edges. GRNsight has general applicability for displaying any small, unweighted or weighted network with directed edges for systems biology or other application domains. GRNsight serves as an example of following and teaching best practices for scientific computing and complying with FAIR Principles, using an open and test-driven development model with rigorous documentation of requirements and issues on GitHub. An exhaustive unit testing framework using Mocha and the Chai assertion library consists of around 160 automated unit tests that examine nearly 530 test files to ensure that the program is running as expected. The GRNsight application (<a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>) and code (<a href="https://github.com/dondi/GRNsight">https://github.com/dondi/GRNsight</a>) are available under the open source BSD license.
                                </p>

                            </div>
                        </div>
                    </div>

                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <p class="panel-title aboutHeaders">
                                <a href="#presentations" data-toggle="collapse">
                                    Presentations
                                </a>
                            </p>
                        </div>

                        <div id="presentations" class="panel-collapse collapse">
                            <div class="panel-body">
                                
                                <p class="abstract">
                                    <b>LMU Department of Biology Seminar Series</b><br><br>
                                    September 16, 2016<br>
                                    GRNmap and GRNsight: Using the power of genomics, mathematics,and open source visualization software to understand gene regulatory networks in yeast<br>
                                    <b>Kam D. Dahlquist</b><br><br>

                                </p>
<br>
                                
                                <p class="abstract">
                                    <b><a href="https://www.open-bio.org/wiki/BOSC_2016">Bioinformatics Open Source Conference (BOSC) 2016, Orlando, FL</a></b><br><br>
                                    July 9, 2016<br>
                                    GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks<br>
                                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Trixie Anne M. Roque, Anindita Varshneya, Mihir Samdarshi, and Chukwuemeka Azinge<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Over a period of several years, our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function.  The large number of developers and time span of development led to a code base that was difficult to revise and adjust. We therefore brought the code under version control in a GitHub repository and refactored the script-based software with global variables into a function-based package that uses an object to carry relevant information from function to function. This modular approach allows for cleaner, less ambiguous code and increased maintainability. We standardized the format of the input and output Excel workbooks, making them more readable. We also added an optimization diagnostics output worksheet which includes both the actual and theoretical minimum least squared error overall, and the mean squared errors for the individual genes. The MATLAB compiler was used to create an executable that can be run on any Windows machine without the need of a MATLAB license, increasing the accessibility of our program. Finally, we have implemented test-driven development, creating unit tests for all new features to speed up debugging and to prevent future code regressions.  We are improving the test coverage of previous code.<br><br>
                                    GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. It is written in JavaScript, with diagrams facilitated by D3.js. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized. GRNsight uses pointed and blunt arrowheads, and colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. Visualizations can be modified through manual node dragging and sliders that adjust the force graph parameters. From the early stages, GRNsight has had a unit testing framework using Mocha and the Chai assertion library to perform test-driven development where unit tests are written before new functionality is coded. This framework consists of over 160 automated unit tests that examine over 450 test files to ensure that the program is running as expected. Error and warning messages inform the user what happened, the source of the problem, and possible solutions. <br><br>
                                    Together, the life cycle of these two programs illustrate the differences between the cultures of mathematics and computing, the challenges and benefits of bringing an existing code base up to open development standards (GRNmap), and the advantages of starting a project using best practices from the beginning (GRNsight).  Our goal is to facilitate reproducible research.<br><br>
                                     <a href="http://f1000research.com/slides/5-1637">Click here to view the slides on F1000Research.</a><br><br>
                                     
                                     <iframe src="http://www.slideshare.net/GRNsight/slideshelf" width="615px" height="470px" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:none;" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>
                                     
                                     <!--<figcaption><a href="http://www.slideshare.net/GRNsight/dahlquist-bosc-20160709-64224474" target="_blank">Slides</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>-->
                                </p>
<br>
                                <p class="abstract">
                                    <b><a href="http://experimentalbiology.org/PDFS/2016/EB-2016-Program-Web-FullProgram.aspx">Experimental Biology 2016, San Diego, CA</a></b><br><br>
                                    April 4, 2016<br>
                                    <a href="http://www.fasebj.org/content/30/1_Supplement/819.16.abstract?sid=a80e9e6c-998f-4889-b1fd-ddcc75ab8bda">GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks</a><br>
                                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Monica V. Hong, Kristen M. Horstmann, Kayla C. Jackson, K. Grace Johnson, Tessa A. Morris, Trixie Anne M. Roque, Mihir Samdarshi, Anindita Varshneya, Natalie E. Williams, and Kevin W. Wyllie<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function. Parameters for a series of related networks, ranging in size from 15 to 35 genes, were optimized against DNA microarray data measuring the transcriptional response to cold shock in wild type and five strains individually deleted for the transcription factors, Cin5, Gln3, Hap4, Hmo1, Zap1, of budding yeast, Saccharomyces cerevisiae BY4741. Model predictions fit the experimental data well, within the 95% confidence interval. Open source code and a compiled executable that can run without a MATLAB license are available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>. GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. The GRNsight open source code and application are available from <a href="http://dondi.github.io/GRNsight/index.html">http://dondi.github.io/GRNsight/index.html</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/key/NC3auEnK2HuL1f" width="595" height="485" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>

                                    <figcaption><a href="//www.slideshare.net/GRNsight/dahlquist-experimental-biology20160404" title="Dahlquist experimental biology_20160404" target="_blank">Presentation slides</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b><a href="http://ccbs.uci.edu/scsb2015home">5<sup>th</sup> Annual Southern California Systems Biology Conference</a></b><br><br>
                                    January 31, 2015<br>
                                    GRNmap and GRNsight: Open Source Software for Dynamical Systems Modeling and Visualization of Medium-Scale Gene Regulatory Networks<br>
                                    <b>Kam D. Dahlquist</b>,  Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Nicholas A. Rohacz, Katrina Sherbina, Britain J. Southwick, and Anindita Varshneya<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. Parameters for a 21-gene network were optimized against DNA microarray data measuring the transcriptional response to cold shock in wild type and four transcription factor deletion strains of budding yeast, Saccharomyces cerevisiae. Model predictions fit experimental data well, within the 95% confidence interval. Open source code and a compiled executable that can run without a MATLAB license are available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>. GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. The GRNsight code and application are available from <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/index.html</a>. This work was partially supported by NSF award 0921038.
                                </p>

                                <figure class="publication">
                                    <iframe src="http://www.slideshare.net/slideshow/embed_code/44236648" width="425" height="355" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>
                                    <figcaption><a href="http://www.slideshare.net/GRNsight/dahlquist-so-calsysbio20140131" title="Dahlquist so calsysbio_20140131" target="_blank">Presentation slides</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b>Loyola Marymount University CMSI 402 Presentation</b><br><br>
                                    May 8, 2014<br>
                                    GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Britain Southwick</b>
                                </p>

                                <figure class="publication">
                                    <iframe src="http://www.slideshare.net/slideshow/embed_code/35811802" width="427" height="356" allowfullscreen></iframe>

                                    <figcaption><a href="https://www.slideshare.net/GRNsight/southwick-britain-grnsightcmsi402presentation20140508" title="Southwick britain gr_nsight_cmsi402-presentation_20140508" target="_blank">Presentation slides</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b><a href="http://academics.lmu.edu/media/lmuacademics/undergraduateresearchsymposium/documents/Full%20Program%20with%20abstracts%203.pdf">Loyola Marymount University 6<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                                    March 29, 2014<br>
                                    GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Britain Southwick, Nicole Anguiano</b><br>
                                    John David N. Dionisio, Kam D. Dahlquist, Ben G. Fitzpatrick<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them, which govern the level of expression of mRNA and protein from those genes. GRNs can be mathematically modeled and simulated by applications such as GRNmap, a MATLAB program that estimates the parameters and performs forward simulations of a differential equations model of a GRN. Computer representations of GRNs, such as the models output by GRNmap, are tabular spreadsheets that are not easily interpretable. Ideally, GRNs should be displayed as diagrams (graphs) detailing the regulatory relationships (edges) between each gene (node) in the network. To address this need, we developed GRNsight. GRNsight is an open source web application for visualizing mathematical models of GRNs. It allows users to upload spreadsheets generated by GRNmap and uses the information in these spreadsheets to automatically create and display a graph of the GRN model. The application colors the edges and adjusts their thickness based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Finally, GRNsight then allows the user to modify the graph in order to define the best visual layout for the network. Most of GRNsight is written in JavaScript. HTTP requests are handled using Node.js and the Express framework. Graphs are generated through D3.js, a JavaScript data visualization library. <br>
                                </p>

                                <figure class="publication">
                                    <iframe src="http://www.slideshare.net/slideshow/embed_code/35811669" width="427" height="356" allowfullscreen></iframe>
                                    <figcaption><a href="https://www.slideshare.net/GRNsight/southwick-anguiano-lmusymposiumpresentation20140329" title="Southwick anguiano lmu-symposium_presentation_20140329" target="_blank">Presentation slides</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>
                            </div>
                        </div>
                    </div>

                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <p class="panel-title aboutHeaders">
                                <a href="#posters" data-toggle="collapse">
                                    Posters
                                </a>
                            </p>
                        </div>

                        <div id="posters" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p class="abstract">
                                    <b><a href="https://www.open-bio.org/wiki/BOSC_2016">Bioinformatics Open Source Conference (BOSC) 2016</a> and <a href="http://www.iscb.org/ismb2016">Intelligent Systems for Molecular Biology (ISMB) 2016</a>, Orlando, FL</a></b><br><br>
                                    July 8-12, 2016<br>
                                    GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks<br>
                                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Tessa A. Morris, Anindita Varshneya, Natalie E. Williams, K. Grace Johnson, Trixie Anne M. Roque, Kristen M. Horstmann, Mihir Samdarshi, Chukwuemeka E. Azinge, Brandon J. Klein, Margaret J. O’Neil1<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our open source MATLAB software package, GRNmap (<a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>), uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function.  Parameters for a series of related networks, ranging in size from 15 to 35 genes, were optimized against DNA microarray data measuring the transcriptional response to cold shock in budding yeast, Saccharomyces cerevisiae, for the wild type strain and strains deleted for the transcription factors Cin5, Gln3, Hap4, Hmo1, and Zap1, giving biological insights into this process. GRNsight is an open source web application for visualizing such models of gene regulatory networks (<a href="http://dondi.github.io/GRNsight/index.html">http://dondi.github.io/GRNsight/index.html</a>). GRNsight accepts GRNmap- or user-generated Excel spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. This work was partially supported by NSF award 0921038.<br><br>
                                     <a href="http://f1000research.com/posters/5-1618">Click here</a> to view the poster on F1000Research. <a href="//www.slideshare.net/GRNsight/dahlquist-etal-boscismb2016poster-64224013" title="Dahlquist et-al bosc-ismb_2016_poster" target="_blank">Click here</a> to view poster from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare.<br><br>

                                </p>
                                <p class="abstract">
                                    <b><a href="http://experimentalbiology.org/PDFS/2016/EB-2016-Program-Web-FullProgram.aspx">Experimental Biology 2016, San Diego, CA</a></b><br><br>
                                    April 4, 2016<br>
                                    <a href="http://www.fasebj.org/content/30/1_Supplement/819.10.abstract?sid=78e46198-a079-4985-9ebd-0e286b8dde38">Test-driven development improves GRNsight: a web application for visualizing models of gene regulatory networks</a><br>
                                    <b>Anindita Varshneya, Mihir Samdarshi</b>, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN, can mathematically model the dynamics of GRNs. GRNsight automatically lays out the network graph based on GRNmap output spreadsheets. GRNsight uses pointed and blunt arrowheads, and colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. Visualizations can be modified through manual node dragging and sliders that adjust the force graph parameters. We have now implemented an exhaustive unit testing framework using Mocha and the Chai assertion library to perform test-driven development where unit tests are written before new functionality is coded. This framework consists of over 135 automated unit tests that examine about 450 test files to ensure that the program is running as expected. Error and warning messages have a three-part framework that informs the user what happened, the source of the problem, and possible solutions. For example, GRNsight returns an error when the spreadsheet is formatted incorrectly or the maximum number of nodes or edges is exceeded. The completion of the testing framework marks the close of development for version 1 (the current release stands at version 1.12). In version 2.0 of GRNsight, a new feature will be implemented that colors the nodes (genes) based on the expression data. The user will have the choice to display the experimental data or the data produced by the forward simulation feature in GRNmap. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/key/neLFcr6hwpjpkT" width="595" height="485" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>

                                    <figcaption><a href="//www.slideshare.net/GRNsight/varshneya-samdarshi-lmusymposium2016" title="Varshneya samdarshi lmu_symposium_2016" target="_blank">Poster</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>
<br><br>
                                <p class="abstract">
                                    <b><a href="http://academics.lmu.edu/media/lmuacademics/undergraduateresearchsymposium/documents/ei8ht_Symposium_Long_Program_16_Secured.pdf">Loyola Marymount University 8<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                                    March 19, 2016<br>
                                    Test-driven development improves GRNsight: a web application for visualizing models of gene regulatory networks<br>
                                    <b>Anindita Varshneya, Mihir Samdarshi</b>, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is an open source web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN, can mathematically model the dynamics of GRNs. GRNsight automatically lays out the network graph based on GRNmap output spreadsheets. GRNsight uses pointed and blunt arrowheads, and colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. Visualizations can be modified through manual node dragging and sliders that adjust the force graph parameters. We have now implemented an exhaustive unit testing framework using Mocha and the Chai assertion library to perform test-driven development where unit tests are written before new functionality is coded. This framework consists of over 160 automated unit tests that examine over 450 test files to ensure that the program is running as expected. Error and warning messages inform the user what happened, the source of the problem, and possible solutions. The completion of the testing framework marks the close of development for version 1 (the current release stands at version 1.12). In version 2.0 of GRNsight, a new feature will be implemented that colors the nodes (genes) based on expression data provided by GRNmap. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/key/neLFcr6hwpjpkT" width="595" height="485" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>

                                    <figcaption><a href="//www.slideshare.net/GRNsight/varshneya-samdarshi-lmusymposium2016" title="Varshneya samdarshi lmu_symposium_2016" target="_blank">Poster</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b><a href="http://www.pointloma.edu/experience/academics/schools-departments/department-biology/science-education-programs/wcbsurc/40th-annual-wcbsurc-plnu-2015">40<sup>th</sup> Annual West Coast Biological Sciences Undergraduate Research Conference</a></b><br><br>
                                    April 25, 2015 <br>
                                    Improvements to GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Nicole Anguiano, Anindita Varshneya</b>, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRN). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN, can mathematically model GRNs. The general graph visualization tools that exist for displaying GRNs did not fulfill our needs, so we created GRNsight. GRNsight automatically lays out the network graph based on spreadsheets uploaded from GRNmap. GRNsight colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. GRNsight uses pointed and blunt arrowheads to indicate activation and repression, respectively. Visualizations can be modified through manual node dragging and sliders that adjust the force layout parameters. Since version 1.0, GRNsight has acquired significant user interface improvements, bug fixes, extended robustness to errors, and aesthetic refinements. Demo spreadsheets are now available for users who do not have their own models. GRNsight's current release is version 1.10, with beta version 1.11 available for testing new features. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="http://www.slideshare.net/slideshow/embed_code/46240444" width="476" height="400" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>

                                    <figcaption><a href="http://www.slideshare.net/GRNsight/anguiano-varshneya-lmusymposium2015" title="Anguiano varshneya lmu_symposium_2015" target="_blank">Poster</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b><a href="http://academics.lmu.edu/media/lmuacademics/undergraduateresearchsymposium/documents/Final%20Program%20w.%20Abstracts.pdf">Loyola Marymount University 7<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                                    March 21, 2015<br>
                                    Improvements to GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Nicole Anguiano, Anindita Varshneya</b>, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRN). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNs can be mathematically modeled by GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN. The general graph visualization tools that exist for displaying GRNs did not fulfill our needs, so we created GRNsight. GRNsight automatically lays out the network graph based on spreadsheets uploaded from GRNmap. It is written in JavaScript, with diagrams facilitated by D3.js. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized to support the specific needs of GRN visualization. GRNsight colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. GRNsight uses pointed and blunt arrowheads to indicate activation and repression, respectively. Visualizations can be modified through manual node dragging and sliders that adjust D3.js’s force layout parameters. Since version 1.0, GRNsight has acquired significant user interface improvements, bug fixes, extended robustness to errors in file formats, and aesthetic refinements, particularly for edge markers. Demo spreadsheets are now available for users who do not have their own models. GRNsight's current release is version 1.8, with beta version 1.9 available for testing new features. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="http://www.slideshare.net/slideshow/embed_code/46240444" width="476" height="400" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>

                                    <figcaption><a href="http://www.slideshare.net/GRNsight/anguiano-varshneya-lmusymposium2015" title="Anguiano varshneya lmu_symposium_2015" target="_blank">Poster</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</div>
                                </figure>

                                <p class="abstract">
                                    <b><a href="http://ccbs.uci.edu/scsb2015home">5<sup>th</sup> Annual Southern California Systems Biology Conference</a></b><br><br>
                                    January 31, 2015<br>
                                    Improvements to GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Nicole Anguiano</b>, Anindita Varshneya, Britain J. Southwick, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRN). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNs can be mathematically modeled by applications such as GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN. GRNsight automatically lays out the network graph based on spreadsheets uploaded from GRNmap. It is written in JavaScript, with diagrams facilitated by D3.js. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized to support the specific needs of GRN visualization. GRNsight colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. GRNsight uses pointed and blunt arrowheads to indicate activation and repression, respectively. When a user mouses over an edge, the numerical value of the weight parameter is displayed. Visualizations can be modified through manual node dragging and sliders that adjust D3.js’s force layout parameters. Since version 1.0, GRNsight has acquired significant user interface improvements, bug fixes, extended robustness to errors in file formats, and aesthetic refinements, particularly for edge markers. Demo spreadsheets are now available for users who do not have their own models. GRNsight's current release is version 1.8, with beta version 1.9 available for testing new features. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="http://www.slideshare.net/slideshow/embed_code/42279842" width="476" height="400" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>

                                    <figcaption><a href="http://www.slideshare.net/GRNsight/anguiano-varshneya-sccurposter20141122" title="Anguiano varshneya sccur-poster_20141122" target="_blank">Poster</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b>2014 Southern California Conference for Undergraduate Research</b><br><br>
                                    November 22, 2014<br>
                                    GRNsight: A Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Nicole Anguiano, Anindita Varshneya</b>, John David N. Dionisio, Kam D. Dahlquist, Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRN). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNs can be mathematically modeled by applications such as GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN. The general graph visualization tools that exist for displaying GRNs did not fulfill our needs, so we created GRNsight. GRNsight automatically lays out the network graph based on spreadsheets uploaded from GRNmap. It is written in JavaScript, with diagrams facilitated by D3.js, a data visualization library. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized to support the specific needs of GRN visualization. GRNsight colors the edges and adjusts their thickness based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. GRNsight uses arrowheads and perpendicular line segments to indicate activation and repression, respectively. When a user mouses over an edge, the numerical value of the weight parameter is displayed. Visualizations can be modified through manual node dragging and sliders that adjust D3.js’s force layout parameters. GRNsight follows an open development model with rigorous documentation of requirements and issues on GitHub. Since version one, GRNsight has acquired significant user interface improvements, bug fixes, extended robustness to errors in file formats, and aesthetic refinements, particularly for edge markers. Demo spreadsheets are now available for users who do not have their own models. GRNsight's current release is version 1.6, with beta version 1.7 available for testing new features. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="http://www.slideshare.net/slideshow/embed_code/42279842" width="476" height="400" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>

                                    <figcaption><a href="http://www.slideshare.net/GRNsight/anguiano-varshneya-sccurposter20141122" title="Anguiano varshneya sccur-poster_20141122" target="_blank">Poster</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
